commit 1ea53d4684469e41dd342f992ccbb61a48cd8118
Author: q66 <q66@chimera-linux.org>
Date:   Wed Sep 16 11:20:16 2026 +0200

    fix test-repacker length values for test_32bit_roots_traversal
    
    add_object takes memory for the tag followed by length of the
    memory, in this case the tag has 2 bytes but the length passed
    is 4 by accident which triggers a fortify violation later on

diff --git a/src/test-repacker.cc b/src/test-repacker.cc
index bf0c388..203c990 100644
--- a/src/test-repacker.cc
+++ b/src/test-repacker.cc
@@ -2788,8 +2788,8 @@ test_32bit_roots_traversal ()
   hb_serialize_context_t c (buffer, buffer_size);
 
   c.start_serialize ();
-  unsigned leaf1 = add_object ("l1", 4, &c);
-  unsigned leaf2 = add_object ("l2", 4, &c);
+  unsigned leaf1 = add_object ("l1", 2, &c);
+  unsigned leaf2 = add_object ("l2", 2, &c);
 
   start_object ("mid", 4, &c);
   add_wide_offset (leaf1, &c);
